iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableExtension.java

```diff
@@ -34,13 +33,41 @@ public EventProcessingResponse processEventProcessingRequest(EventProcessingRequ
             String apiKey = account.getStringSetting(SETTING_API_KEY, true, null);
             iterableService = IterableService.newInstance(apiKey);
         }
-        updateUser(new Event.Context(request));
+        Collections.sort(
+                request.getEvents(),
+                (a, b) -> a.getTimestamp() > b.getTimestamp() ? 1 : a.getTimestamp() == b.getTimestamp() ? 0 : -1
+        );
+        insertPlaceholderEmail(request);
+        updateUser(request);
         return super.processEventProcessingRequest(request);
     }
 
+    /**
+     * Verify that there's an email present, create a placeholder if not.
+     *
+     * @param request
+     * @throws IOException
+     */
+    private void insertPlaceholderEmail(EventProcessingRequest request) throws IOException {
+        long count = request.getUserIdentities() == null ? 0 : request.getUserIdentities().stream()
+                .filter(t -> t.getType().equals(UserIdentity.Type.EMAIL))
+                .count();
+        if (count > 0) {
+            return;
+        }
+        String placeholderEmail = getPlaceholderEmail(request);
+        if (request.getUserIdentities() == null) {
```

Comment:
Really minor / nit picky but you check for request.getUserIdentities() == null twice. Could refactor that into a bool of its own.